### ISSUE TITLE
AC0031 CodeFix: quick fix adds only one table when several diagnostics share a location

### ISSUE THREAD

**Issue by Arthurvdv** [ALCops/Analyzers#492]
### Analyzer
ApplicationCop

### Rule ID
AC0031

### Description
`TableDataAccessRequiresPermissionsCodeFixProvider` acts on `ctx.Diagnostics[0]` only. Since #491, a `DataTransfer` executor can report several AC0031 diagnostics at the **same** location for **different** tables (`r` on the source, `m`/`i` on the destination). The interactive lightbulb therefore adds only the first table's permission; the developer has to invoke the fix again for the second table. FixAll (per-diagnostic batch) is unaffected, so the existing HasFix tests do not catch it.

### AL code to reproduce
```al
codeunit 50100 MyUpgrade
{
    Subtype = Upgrade;

    trigger OnRun()
    var
        MyDataTransfer: DataTransfer;
    begin
        MyDataTransfer.SetTables(Database::TableA, Database::TableB);
        MyDataTransfer.AddFieldValue(1, 1);
        MyDataTransfer.CopyFields(); // AC0031 x2: TableA = r, TableB = m
    end;
}
```
Applying the offered fix adds only `tabledata TableA = r`.

### Suggested fix
Register one code action that covers every AC0031 diagnostic in `ctx.Diagnostics` (all share the span), adding/merging one `Permissions` entry per table, and add a HasFix fixture for the two-table DataTransfer case.

Follow-up from the review of #491.
